pkg/util/steps/condition.go

```diff
@@ -66,9 +86,29 @@ func (c conditionStep) run(ctx context.Context, log *logrus.Entry) error {
 		log.Warnf("step %s failed but has configured 'fail=%t'. Continuing. Error: %s", c, c.fail, err.Error())
 		return nil
 	}
+	if err == wait.ErrWaitTimeout {
```

Comment:
I don't think it necessarily needs to be handled in this PR, but I have noticed that we see `ErrWaitTimeout` errors returned in the internal implementation of some condition steps (e.g. a k8s client list call takes too long to resolve). 

I've opened a PR to wrap "internal" ErrWaitTimeout errors within the condition runner [here](https://github.com/Azure/ARO-RP/pull/3005), which will change the shape of the log message we see but preserve the error type as `ErrWaitTimeout`. 

If we want this error enrichment to handle both "the condition timed out" and "something within the condition timed out" cases, this if statement should be changed to `errors.Is(err, wait.ErrWaitTimeout)` instead of the equality comparison, in order to match the wrapped error.

Reply:
Yes, agreed, made the change.